src/pages/api/attest.ts

```diff
@@ -1,19 +1,77 @@
 import { NextApiRequest, NextApiResponse } from 'next';
 import { createAttestation } from '@/helpers/eas';
 import { privateKeyToSigner } from '@/utils/eas-wagmi-utils';
+import {
+  updateScoreRecord,
+  handleReceipt,
+  markReceiptAsUsed,
+} from '@/helpers/databaseHelpers';
+import pinataSDK from '@pinata/sdk';
+import { ATTESTER_PUBLIC_KEY } from '@/constants';
 
 export default async (req: NextApiRequest, res: NextApiResponse) => {
   try {
+    const attesterPvtKey = process.env.ATTESTER_PVT_KEY;
+    if (!attesterPvtKey)
+      return res.status(400).json({ error: 'No attester set' });
     const body = req.body;
-    const signer = privateKeyToSigner(process.env.ATTESTER_PVT_KEY!);
-    const result = await createAttestation(
-      body.address,
-      body.score,
-      body.meta,
-      signer,
-    );
-    return res.status(200).json(result);
-  } catch (e) {
-    return res.status(200).send({ score: 0, meta: {} }); // send an empty debug object on error
+    const { address, score, meta, network, receipt } = body;
+
+    // Check if the receipt has already been used
+    try {
+      await handleReceipt(receipt, address);
+    } catch (error) {
+      return res.status(400).json({ error: error });
+    }
+
+    const { signer, provider } = privateKeyToSigner(attesterPvtKey, network);
+    // check receipt
+    const transaction = await provider.getTransaction(receipt);
+    if (
+      transaction &&
+      transaction.from.toLowerCase() === address.toLowerCase() &&
+      transaction.to?.toLowerCase() === ATTESTER_PUBLIC_KEY.toLowerCase() &&
+      transaction.confirmations > 0
```

Comment:
let's use the eth price function here and verify that the amount the user sent is 1USD or more. Additionally, we need to check the price at the time the tx was sent, because the price might drop by a few cents in the meantime. Let's also introduce a very small tolerance - 0.05 USD - to account for price fluctuation.